lib/Service/PageService.php

```diff
@@ -371,7 +371,9 @@ public function pageHasOtherContent(File $file): bool {
 					!self::isIndexPage($node)) {
 					return true;
 				}
-				if ($node->getName() !== PageInfo::INDEX_PAGE_TITLE . PageInfo::SUFFIX) {
+				// Ignore the page file itself and corresponding attachments folder
+				if ($node->getName() !== PageInfo::INDEX_PAGE_TITLE . PageInfo::SUFFIX
```

Comment:
This could also use the `isIndexPage function, right?
```suggestion
				if (!self::isIndexPage($node))
```

Reply:
I tried to understand this function on its own and noticed it does not work for normal pages - only for index pages because otherwise the `getParent()->getDirectoryListing()` in 368 will fetch all sibling pages.

I see that this is only used on index pages. Still maybe rename to `indexPageHasOtherContent`, document this limitation or add a check
```php
if (!self::isIndexPage($file)) {
  return false;
}
```
before the foreach.

Reply:
Thanks, both valid points. I renamed the function to `indexPageHasOtherContent` and used `self::isIndexPage()` inside it now.